client/my-sites/plans-features-main/hooks/use-is-plan-upsell-enabled-on-free-domain.ts

```diff
@@ -0,0 +1,18 @@
+import { useExperiment } from 'calypso/lib/explat';
+import type { DataResponse } from 'calypso/my-sites/plan-features-2023-grid/types';
+
+export const useIsPlanUpsellEnabledOnFreeDomain = (
+	flowName?: string | null
+): DataResponse< boolean > => {
+	const ONBOARDING_EXPERIMENT = 'calypso_gf_signup_onboarding_free_free_dont_miss_out_modal';
+	const ONBOARDING_PM_EXPERIMENT = 'calypso_gf_signup_onboarding_pm_free_free_dont_miss_out_modal';
+	const relevantExperiment =
+		flowName === 'onboarding' ? ONBOARDING_EXPERIMENT : ONBOARDING_PM_EXPERIMENT;
+	const [ isLoading, experimentAssignment ] = useExperiment( relevantExperiment, {
+		isEligible: [ 'onboarding', 'onboarding-pm' ].includes( flowName ?? '' ),
```

Comment:
IMO, we should only assign users to the experiment if a free domain was selected. Can we pass some param for `isEligible` check so that users who selected a paid domain are not assigned?